Added progress bar to nav bar

llau6 opened the pull request:
- Removed debug logs

![image](https://user-images.githubusercontent.com/10525274/33645199-a32bead4-da17-11e7-8560-656dc3d0b504.png)
- Added progress bar
- Added current location

![image](https://user-images.githubusercontent.com/10525274/33645258-daca1f2e-da17-11e7-8f87-0155f299f250.png)
- Added icon (pen) for accessing Scratch
- Changed social media icon to a globe

bnmnetp commented on runestone/activecode/test/_templates/plugin_layouts/sphinx_bootstrap/static/bootstrap-sphinx.js:
```diff
@@ -52,6 +54,7 @@
   };
 
   $(document).ready(function () {
+    console.log("bootstrap-sphinx.js running")
```
This looks like debug code that should be removed before I accept the PR

bnmnetp commented on runestone/activecode/test/test_activecode.py:
```diff
@@ -5,7 +5,7 @@
 
 setUpModule, tearDownModule = module_fixture_maker(__file__)
-
+print "test_activecode.py"
```
The runestone components are Python3 compatible, so print should use parenthesis.  Is this also debug code that should be removed?

bnmnetp commented on runestone/common/js/navhelp.js:
```diff
@@ -3,7 +3,7 @@
 
 /* Sets up the interactive navhelp */
-
+var navHelp = "this is from navHelp.js"
```
Does this serve a purpose?

bnmnetp commented on runestone/common/project_template/_templates/plugin_layouts/sphinx_bootstrap/globaltoc.html:
```diff
@@ -1,3 +1,6 @@
+<script>
+  console.log("globaltoc");
+</script>
```
More debug?

bnmnetp commented:
@llau6 @ericsonga I like the progress in chapter, progress in book on the right side.  But the left side of chapter / subchapter location has replaced the navigation dropdown that allows you to quickly move between chapters.  

I would like to keep that dropdown, or at least keep it as a build option if you prefer one or the other.

llau6 commented:
@bnmnetp It looks like on the book I was working on (apcsareview) there doesn't exist a Change Dropdown. I went ahead and downloaded thinkcspy and did see the dropdown you mentioned. Do you know why the dropdown wouldn't appear on apcsareview? I figured the answer would be in conf.py under `html_theme_options` as an option. I see that apcsareview has `'navbar_site_nam e': "Chapters"` listed as an option, but there still is no Chapter Dropdown. Do you know if this is how to turn on and off the Chapter Dropdown?

llau6 commented:
@ericsonga @bnmnetp I have removed the debug logs and added the progress bar logic inside its own filed called `progress.html`. Please let me know if you run into any issues that I need to fix.

bnmnetp commented on runestone/activecode/js/activecode.js:
```diff
@@ -61,8 +61,8 @@ ActiveCode.prototype.init = function(opts) {
     this.createEditor();
-    this.createOutput();
-    this.createControls();
+    // this.createOutput();
```
Ugh!  Why were these commented out!??
